File: main/04_replacing_data.py
## Replacing one column
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df['Name'] = ['Argon', 'Balloon', 'Chess', 'Duck', 'Exit']

df.Name = ['Argon', 'Balloon', 'Chess', 'Duck', 'Exit']


## Replacing one row
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df.loc[0] = ['Argon', 1, 'No', False] # not recommended # need to know the exact order of columns


## Replacing a single cell
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df['Name'][0] = 'Argon' # not recommended # pandas warning

df.loc[0, 'Name'] = 'Argon'

df.at[0, 'Name'] = 'Argon'

df.iloc[0, 0] = 'Argon' # note: this may be a different column due to pandas' ordering

df.iat[0, 0] = 'Argon' # see above

File: main/04_replacing_data.R
## Replacing one column
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df$Name = c('Argon', 'Balloon', 'Chess', 'Duck', 'Exit')

df[, 1] = c('Argon', 'Balloon', 'Chess', 'Duck', 'Exit')

df[[1]] = c('Argon', 'Balloon', 'Chess', 'Duck', 'Exit')


## Replacing one row
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df$Name = as.character(df$Name) # necessary to prevent NA in next step
df[1, ] = c('Argon', 1, 'No', FALSE)

# data.table
library(data.table)
dt = fread('00_R_data_set.tsv')
namelist = c('Argon', 'Balloon', 'Chess', 'Duck', 'Exit')
dt[, Name := namelist]


## Replacing a single cell
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df$Name[1] = 'Argon'

df[1, 1] = 'Argon'
